test/generation/test_gen.py

```diff
@@ -318,7 +319,7 @@ def test_gen_candidates_scipy_nan_handling(self):
             test_grad = torch.tensor([0.5, 0.2, float("nan")], **ckwargs)
             # test NaN in grad
             with mock.patch("torch.autograd.grad", return_value=[test_grad]):
-                with self.assertRaisesRegex(RuntimeError, expected_regex):
+                with self.assertRaisesRegex(OptimizationGradientError, expected_regex):
```

Comment:
One thing to consider here is that this could break people's code if they rely on this being a `RuntimeError` - how about subclassing `OptimizationGradientError` from both `BoTorchError` and `RuntimeError`? (this is somehting we could probably consider doing more generally)

Reply:
Good point, I will change it accordingly.